hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/ListKeyHandler.java

```diff
@@ -94,4 +103,40 @@ protected void execute(OzoneClient client, OzoneAddress address)
     }
   }
 
+  private void listKeysInsideVolume(OzoneClient client, OzoneAddress address)
+      throws IOException {
+    String volumeName = address.getVolumeName();
+    OzoneVolume vol = client.getObjectStore().getVolume(volumeName);
+
+    Iterator<? extends OzoneBucket> bucketIterator =
+        vol.listBuckets(null);
+    int maxKeyLimit = listOptions.getLimit();
+    String keyPrefix = "";
+    if (!Strings.isNullOrEmpty(listOptions.getPrefix())) {
+      keyPrefix += listOptions.getPrefix();
+    }
+
+    int totalKeys = 0;
+    while (bucketIterator.hasNext()) {
+      OzoneBucket bucket = bucketIterator.next();
+      Iterator<? extends OzoneKey> keyIterator = bucket.listKeys(keyPrefix,
+          listOptions.getStartItem());
+
+      int counter = printAsJsonArray(keyIterator, maxKeyLimit);
+      totalKeys += counter;
+      maxKeyLimit -= counter;
+
+      // More keys were returned notify about max length
+      if (keyIterator.hasNext() || (bucketIterator.hasNext()
+          && maxKeyLimit <= 0)) {
+        out().println("Listing first " + totalKeys + " entries of the " +
+            "result. Use --length (-l) to override max returned keys.");
+        return;
+      }
+    }
+    if (isVerbose()) {
+      out().printf("Found : %d keys in volume : %s %n",
+          totalKeys, volumeName);
+    }
```

Comment:
Looks like `-l` for the existing list command does not support `-1` as an override to list all keys. It should probably have that, but that can be done in a follow up PR. These two things combined mean the length message will always be there and jq will never work unless -1 is set to something like 100 billion.

Reply:
Filed HDDS-8858 to avoid mixing these messages with the JSON output.  I was thinking about showing only in `--verbose` mode, but `stderr` seems a better idea.

Reply:
The separate Jira sounds good. There are probably other places where we have this problem as well. I guess that makes it optional to fix in this PR or not.

Reply:
> I will create another Jira and handle -1 to override list all keys.

How about adding a `--all` option in `ListOptions`?  If it is set, then `limit` could be set to `MAX_VALUE` or `-1` (whichever works) _internally_.  I think that would be more user-friendly than using a special value publicly.

Reply:
`--all` option sounds good.

Reply:
Sounds good to me as well.